fv3core/grid/mirror.py

```diff
@@ -0,0 +1,342 @@
+import math
+
+from ..utils.global_constants import N_TILES, PI, RADIUS, RIGHT_HAND_GRID
+
+
+__all__ = ["mirror_grid"]
+
+# A tile global version of mirror_grid
+# Closer to the Fortran code
+def global_mirror_grid(grid_global, ng: int, npx: int, npy: int, np):
+    # first fix base region
+    nreg = 0
+    for j in range(0, math.ceil(npy / 2)):
+        for i in range(0, math.ceil(npx / 2)):
+            x1 = 0.25 * (
+                np.abs(grid_global[ng + i, ng + j, 0, nreg])
+                + np.abs(grid_global[ng + npx - (i + 1), ng + j, 0, nreg])
+                + np.abs(grid_global[ng + i, ng + npy - (j + 1), 0, nreg])
+                + np.abs(grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 0, nreg])
+            )
+            grid_global[ng + i, ng + j, 0, nreg] = np.copysign(
+                x1, grid_global[ng + i, ng + j, 0, nreg]
+            )
+            grid_global[ng + npx - (i + 1), ng + j, 0, nreg] = np.copysign(
+                x1, grid_global[ng + npx - (i + 1), ng + j, 0, nreg]
+            )
+            grid_global[ng + i, ng + npy - (j + 1), 0, nreg] = np.copysign(
+                x1, grid_global[ng + i, ng + npy - (j + 1), 0, nreg]
+            )
+            grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 0, nreg] = np.copysign(
+                x1, grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 0, nreg]
+            )
+
+            y1 = 0.25 * (
+                np.abs(grid_global[ng + i, ng + j, 1, nreg])
+                + np.abs(grid_global[ng + npx - (i + 1), ng + j, 1, nreg])
+                + np.abs(grid_global[ng + i, ng + npy - (j + 1), 1, nreg])
+                + np.abs(grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 1, nreg])
+            )
+
+            grid_global[ng + i, ng + j, 1, nreg] = np.copysign(
+                y1, grid_global[ng + i, ng + j, 1, nreg]
+            )
+            grid_global[ng + npx - (i + 1), ng + j, 1, nreg] = np.copysign(
+                y1, grid_global[ng + npx - (i + 1), ng + j, 1, nreg]
+            )
+            grid_global[ng + i, ng + npy - (j + 1), 1, nreg] = np.copysign(
+                y1, grid_global[ng + i, ng + npy - (j + 1), 1, nreg]
+            )
+            grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 1, nreg] = np.copysign(
+                y1, grid_global[ng + npx - (i + 1), ng + npy - (j + 1), 1, nreg]
+            )
+
+            # force dateline/greenwich-meridion consitency
+            if npx % 2 != 0:
+                # TODO: this seems to not make a difference
```

Comment:
This sounds like a TODO that won't be resolved after you merge this. Can you resolve it now?

Reply:
It looks like the idea is to ensure that the longitude at the center of tile 0 is exactly 0 and not off by some rounding error. It isn't an issue in the test cases we're running, but I could see a different grid resolution having issues here. Maybe that means we need to test at a different resolution to ensure this gets covered, but I'm comfortable leaving it in.